backend/src/main/scala/com/softwaremill/adopttapir/starter/formatting/ProjectFormatter.scala

```diff
@@ -0,0 +1,51 @@
+package com.softwaremill.adopttapir.starter.formatting
+
+import cats.effect.IO
+import com.softwaremill.adopttapir.logging.FLogging
+import com.softwaremill.adopttapir.starter.files.FilesManager
+import com.softwaremill.adopttapir.template.{CommonObjectTemplate, GeneratedFile}
+import org.scalafmt.interfaces.Scalafmt
+
+import java.nio.file.Path
+
+class ProjectFormatter(fm: FilesManager) extends FLogging {
+
+  private val ScalaFileExtensions = Set(".scala", ".sbt")
+
+  private lazy val scalafmt = Scalafmt.create(getClass.getClassLoader)
+
+  def format(gfs: List[GeneratedFile]): IO[List[GeneratedFile]] = {
+    findGeneratedFormatFile(gfs) match {
+      case Some(formatFile) => {
+        IO.blocking(fm.createTempDir())
+          .bracket { tempDirectory =>
+            val formattedFiles = for {
+              tempDir <- tempDirectory
+              formatFile <- fm.createFile(tempDir, formatFile)
+              formattedFile <- IO(formatScalaFiles(formatFile, gfs))
+            } yield formattedFile
+
+            formattedFiles
+          }(release = tempDirectory => fm.deleteFilesAsStatedInConfig(tempDirectory))
+      }
+      case None => {
+        logger.error(s"Cannot find formatting file in generated project, scala files would NOT be formatted!")
+        IO(gfs)
+      }
+    }
+  }
+
+  private def findGeneratedFormatFile(gfs: List[GeneratedFile]): Option[GeneratedFile] = {
+    gfs.find(_.relativePath.endsWith(CommonObjectTemplate.scalafmtConfigPath))
+  }
+
+  private def formatScalaFiles(scalaFormatPath: Path, gfs: List[GeneratedFile]): List[GeneratedFile] = {
+    gfs.map {
+      case gf if isScalaFilePath(gf.relativePath) =>
```

Comment:
nit: naming - `gf` would be self explanatory if it was turned to either `generatedFile` or just simply `file`